src/anomalib/pre_processing/pre_processing.py

```diff
@@ -0,0 +1,227 @@
+        >>> from anomalib.pre_processing import PreProcessor
+
+        >>> # Define transforms
+        >>> train_transform = Compose([Resize((224, 224)), ToTensor()])
+        >>> val_transform = Compose([Resize((256, 256)), CenterCrop((224, 224)), ToTensor()])
+
+        >>> # Create PreProcessor with stage-specific transforms
+        >>> pre_processor = PreProcessor(
+        ...     train_transform=train_transform,
+        ...     val_transform=val_transform
+        ... )
+
+        >>> # Create PreProcessor with a single transform for all stages
+        >>> common_transform = Compose([Resize((224, 224)), ToTensor()])
+        >>> pre_processor_common = PreProcessor(transform=common_transform)
+
+        >>> # Use in a Lightning module
+        >>> class MyModel(LightningModule):
+        ...     def __init__(self):
+        ...         super().__init__()
+        ...         self.pre_processor = PreProcessor(...)
+        ...
+        ...     def configure_callbacks(self):
+        ...         return [self.pre_processor]
+        ...
+        ...     def training_step(self, batch, batch_idx):
+        ...         # The pre_processor will automatically apply the correct transform
+        ...         processed_batch = self.pre_processor(batch)
+        ...         # Rest of the training step
+    """
+
+    def __init__(
+        self,
+        train_transform: Transform | None = None,
+        val_transform: Transform | None = None,
+        test_transform: Transform | None = None,
+        transform: Transform | None = None,
+    ) -> None:
+        super().__init__()
+
+        if transform and any([train_transform, val_transform, test_transform]):
+            msg = (
+                "`transforms` cannot be used together with `train_transform`, `val_transform`, `test_transform`.\n"
+                "If you want to apply the same transform to the training, validation and test data, "
+                "use only `transforms`. \n"
+                "Otherwise, specify transforms for training, validation and test individually."
+            )
+            raise ValueError(msg)
+
+        self.train_transform = train_transform or transform
+        self.val_transform = val_transform or transform
+        self.test_transform = test_transform or transform
+        self.predict_transform = self.test_transform
+        self.exportable_transform = get_exportable_transform(self.test_transform)
+
+    def setup_datamodule_transforms(self, datamodule: "AnomalibDataModule") -> None:
+        """Set up datamodule transforms."""
+        # If PreProcessor has transforms, propagate them to datamodule
+        if any([self.train_transform, self.val_transform, self.test_transform]):
+            transforms = {
+                "train": self.train_transform,
+                "val": self.val_transform,
+                "test": self.test_transform,
+            }
+            set_datamodule_transforms(datamodule, transforms)
+            return
```

Comment:
I have the feeling this can be simplified. Can we skip the `set_datamodule_transforms` method, and just call `set_datamodule_stage_transform` directly, while looping over the `train`, `val`, 'test' and `predict` transforms?

Reply:
Removed in f5fbb7c78f4a733e56c9a652d73aaf96014590c4